node/resolvers/profile/services.ts

```diff
@@ -104,27 +109,33 @@ export async function updateProfile(
     .then(() => getProfile(context, extraFields && extraFields.customFieldsStr))
 }
 
-export async function updateProfilePicture(context: Context, file: string) {
+export async function updateProfilePicture(context: Context, file: any) {
   const {
     dataSources: { profile },
     vtex: { currentProfile },
   } = context
 
-  const field = 'profilePicture'
+  const { profilePicture } = await profile.getProfileInfo(
+    currentProfile.email,
+    'profilePicture'
+  )
 
-  const { id } = await profile.getProfileInfo(currentProfile.email, 'id')
+  if (profilePicture) {
+    try {
+      deleteFile(context.vtex, profilePicture)
+    } catch (e) {
+      console.error(e)
```

Comment:
If an error occurs here, it will enter the catch and continue executing the function. So, it will try to updaloadFile below. Wouldn't it be better to let the throw the error?

Reply:
Yes, but no. It's more like an optimization

Reply:
what ? why optimization ?